### ISSUE TITLE
Pre-trained weights of RepLKNet-13 for visualizing the ERF

### ISSUE THREAD

**Issue by ShunLu91** [DingXiaoH/RepLKNet-pytorch#15]
Thanks for your nice work and could you please upload the pre-trained weights of the RepLKNet-13 for us to visualize the ERF?

Best wishes and great thanks!

**Comment by DingXiaoH** [comment]
Sure, we are uploading it!

Regards,
Xiaohan Ding




***@***.***
 
From: ShunLu91
Date: 2022-03-31 17:41
To: DingXiaoH/RepLKNet-pytorch
CC: Subscribed
Subject: [DingXiaoH/RepLKNet-pytorch] Pre-trained weights of RepLKNet-13 for visualizing the ERF (Issue #15)
Thanks for your nice work and could you please upload the pre-trained weights of the RepLKNet-13 for us to visualize the ERF?
Best wishes and great thanks!

**Comment by ShunLu91** [comment]
Thank you very much for your timely reply.

Based on your provided code and pre-trained weights, I visualize the ERF of ResNet-101, ResNet-152 and RepLKNet-31. 
When I loaded the pre-trained weights, the ERF is similar to the figure in your paper. 
However, when they were visualized without pre-trained weights, it appeared a distinct phenomenon that ResNet-101 and ResNet-152 got much larger ERF than RepLKNet-31. 
According to the above observations, I wonder if we can conclude that these networks naturally have a large receptive field but their final distinct ERF is caused by the training rather than the natural ability. 
![image](https://user-images.githubusercontent.com/37992509/161188027-995f9310-07b7-493c-9507-ba37074b9111.png)

Best regards, 
Shun Lu

**Comment by DingXiaoH** [comment]
Thank you for sharing the results and thoughts. We also found that the ERF of RepLKNet expands and the training proceeds. We agree that the ERF visualized at the very beginning may differ from the well-trained models, but it is worth considering **whether the "receptive field" without training makes sense**. I mean, since the model cannot make reasonable predictions before the training begins, the contribution matrix seems not a reasonable metric for the pixel contributions.

**Comment by ShunLu91** [comment]
OK, thanks again for your response and the great contribution to the community. 
I will keep following  dalao's work.
